### ISSUE TITLE
Notification icon on Android 5.1.1

### ISSUE THREAD

**Issue by yarcub** [phonegap-build/PushPlugin#581]
I found an issue after updating Android to 5.1.1.

In Android 5.1.1 the notification icon it's not applied in the notification list nor the status bar, we only see a white icon. Although, it's not the case when the screen is locked. There, the icon appears fine.
![Imgur](http://i.imgur.com/8iRAIEt.jpg)

**Comment by mgrubinger** [comment]
hi @yarcub,
here's what worked for me after hours of trial-and-error:

1. make sure you have the correct size icons. I used this generator: https://romannurik.github.io/AndroidAssetStudio/icons-notification.html

2. place the generated icons/folders in yourapp/platforms/android/res/

3. open yourapp/platforms/android/src/com/plugin/gmc/GCMIntentService.java
Import your package at the top like this
``` import your.package.name.* ```
and replace the line (approx 106)
``` .setSmallIcon(context.getApplicationInfo().icon) ```
with
``` .setSmallIcon(your.package.name.R.drawable.your_notification_icon_filename) ```
e.g.
``` .setSmallIcon(com.yarcub.app.R.drawable.notification_icon) ```

4. Refresh and Clean your project

These posts were helpful to me (besides discussions in the issues of this repo):
http://stackoverflow.com/questions/20596213/white-notification-icon-for-android-with-phonegap-build-and-pushplugin
http://stackoverflow.com/questions/29513478/push-notification-shows-default-icon-instead-of-the-icon-i-setup-in-cordova-conf

hope that helps.

This should be documented somewhere in this repository, it seems many people have this issue. What's your opinion on this @EddyVerbruggen ?

**Comment by EddyVerbruggen** [comment]
Hi,

I don't maintain this repo, but I've implemented that feature in [the fork I'm maintaining](https://github.com/Telerik-Verified-Plugins/PushNotification).

**Comment by mgrubinger** [comment]
Oh, I misread the history of this repo then.
I'll have a look at your repository. thx.

**Comment by yarcub** [comment]
@mgrubinger, unfortunately that solution doesn't fit my needs. We use PhonegapBuild and cannot change platform native code.

@EddyVerbruggen, is the fork compatible with non-Telerik apps?

**Comment by EddyVerbruggen** [comment]
@yarcub Absolutely, that's one of our goals. The idea is to remain compatible but make it easier for Telerik Platform users.

**Comment by ggiordan** [comment]
At the risk of stating the obvious, this for seems dead and it's time to move to @EddyVerbruggen 's fork https://github.com/Telerik-Verified-Plugins/PushNotification?

The plugin itself has been / is super to use... it's time to move to one that is maintained.

Perhaps the better question is, is it time to just move off phonegap and to cordova proper? (assuming I don't use their build service, which I don't)

**Comment by albertleao** [comment]
is @EddyVerbruggen fork a drop in replacement for this one?

**Comment by EddyVerbruggen** [comment]
@albertleao yes, we've tried to remain compatibility with this one.

**Comment by antonkuskov** [comment]
RU: 
Parse sdk сам обрабатывает нотификацию и оно не знает ни о какой иконке. Смотри в [ParsePushBroadcastReceiver.java](https://github.com/ParsePlatform/Parse-SDK-Android/blob/master/Parse/src/main/java/com/parse/ParsePushBroadcastReceiver.java) onPushReceive(). Чтобы показалось иконка твоего приложения, нужно обязательно установить поле "action" у нотификации, создать свой BroadcastReceiver и правильно обработать нотификацию: добавить иконку и прочее..

EN(sorry for my en):
Parse will proccess notification by itself if you will not set field "action". See onPushReceive() in [ParsePushBroadcastReceiver.java](https://github.com/ParsePlatform/Parse-SDK-Android/blob/master/Parse/src/main/java/com/parse/ParsePushBroadcastReceiver.java). To customize notification you have to set "action" field, create your BroadcastReceiver and show notification by using NotificationManager.

**Comment by emrecoban** [comment]
@mgrubinger very very very thank you...
